### PR TITLE
fix: Sytem user permission feature

### PR DESCRIPTION
**Pull request by Aravind-2023**
# Description

made system user information non editable.

Fixes https://github.com/devtron-labs/devtron/issues/1078

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A :  tried to edit system user but not able to do it after my code changes,tested locally

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,8 +47,8 @@
     "scripts": {
         "lint": "eslint src/**/*.tsx",
         "lint-fix": "eslint --fix src/**/*.tsx",
```
Comment (shivani170) [inline package.json:49]: Please revert the changes

File: src/components/userGroups/UserGroup.tsx [hunk 2]
```diff
@@ -639,30 +639,32 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
                 </span>
                 <span
                     className="user-list__direction-container flex rotate pointer"
-                    onClick={email_id === 'admin' ? noop : (e) => setCollapsed(not)}
+                    onClick={(email_id === 'admin'||email_id==='system') ? noop : (e) => setCollapsed(not)}
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:642]: Extract out this inline function as  individual function, Keep admin and system in constant

File: src/setupProxy.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ module.exports = function (app) {
     app.use(
         '/orchestrator',
         createProxyMiddleware({
-            target: 'http://demo.devtron.info:32080/',
+            target: 'http://localhost:8080/',
```
Comment (shivani170) [inline src/setupProxy.js:7]: please revert these changes

File: yarn.lock [hunk 4]
```diff
@@ -4139,9 +4139,9 @@ caniuse-api@^3.0.0:
     lodash.uniq "^4.5.0"
 
 caniuse-lite@^1.0.0, caniuse-lite@^1.0.30000981, caniuse-lite@^1.0.30000989, caniuse-lite@^1.0.30001109, caniuse-lite@^1.0.30001317:
-  version "1.0.30001325"
-  resolved "https://registry.npmjs.org/caniuse-lite/-/caniuse-lite-1.0.30001325.tgz"
-  integrity sha512-sB1bZHjseSjDtijV1Hb7PB2Zd58Kyx+n/9EotvZ4Qcz2K3d0lWB8dB4nb8wN/TsOGFq3UuAm0zQZNQ4SoR7TrQ==
+  version "1.0.30001450"
```
Comment (shivani170) [inline yarn.lock:4142]: Please revert these changes

File: src/components/userGroups/UserGroup.tsx [hunk 5]
```diff
@@ -627,6 +627,14 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
         setData((state) => ({ ...state, result: data }))
         updateCallback(index, data)
     }
+    const decision=email_id==="admin"||email_id==="system";
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:630]: Please rename the `decision` to  somewhat like eg, `isEmaiIIdAdminOrSystem`

File: src/components/userGroups/UserGroup.tsx [hunk 6]
```diff
@@ -639,30 +647,32 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
...
-                                content="Admin user cannot be edited"
+                                content={`${email_id === 'admin' ? 'Admin' : 'System'} user cannot be edited`}
                             >
                                 <div className="flex">{children}</div>
                             </Tippy>
                         )}
+                        
                     >
-                        {email_id === 'admin' ? (
+                        {decision? (
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:667]: It will be true in case email is `system` but here we need to check for only `àdmin`

File: src/setupProxy.js [hunk 7]
```diff
@@ -20,3 +20,4 @@ module.exports = function (app) {
         }),
     )
 }
+//'http://demo.devtron.info:32080/'
```
Comment (shivani170) [inline src/setupProxy.js:23]: Please revert this file

File: yarn.lock [hunk 8]
```diff
@@ -4139,7 +4139,7 @@ caniuse-api@^3.0.0:
     lodash.uniq "^4.5.0"
 
 caniuse-lite@^1.0.0, caniuse-lite@^1.0.30000981, caniuse-lite@^1.0.30000989, caniuse-lite@^1.0.30001109, caniuse-lite@^1.0.30001317:
-  version "1.0.30001325"
+version "1.0.30001325"
```
Comment (shivani170) [inline yarn.lock:4142]: please ignore this change

File: src/components/userGroups/UserGroup.tsx [hunk 9]
```diff
@@ -639,30 +647,32 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
...
                             <Lock />
                         ) : dataLoading ? (
                             <Progressing />
                         ) : (
-                            <NavigationArrow className="arrow-svg" />
+                            <NavigationArrow className="tippyg" />
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:672]: What exactly `tippyg` doing?

File: src/components/userGroups/UserGroup.tsx [hunk 10]
```diff
@@ -639,30 +647,32 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
                 </span>
                 <span
                     className="user-list__direction-container flex rotate pointer"
-                    onClick={email_id === 'admin' ? noop : (e) => setCollapsed(not)}
+                    onClick={onClickHandler}
                     style={{ ['--rotateBy' as any]: collapsed ? '0deg' : '180deg' }}
                 >
                     <ConditionalWrap
-                        condition={email_id === 'admin'}
+                        condition={decision}
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:654]: Condition will be true in case email is `system` but here we need to check for only `àdmin`

File: src/components/userGroups/UserGroup.tsx [hunk 11]
```diff
@@ -627,6 +627,14 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
         setData((state) => ({ ...state, result: data }))
         updateCallback(index, data)
     }
+    const decision=email_id==="admin"||email_id==="system";
+    function onClickHandler(e){
+        if(decision)
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:632]: Please use curlybraces

File: src/components/userGroups/UserGroup.tsx [hunk 12]
```diff
@@ -639,30 +647,32 @@ const CollapsedUserOrGroup: React.FC<CollapsedUserOrGroupProps> = ({
                 </span>
                 <span
                     className="user-list__direction-container flex rotate pointer"
-                    onClick={email_id === 'admin' ? noop : (e) => setCollapsed(not)}
+                    onClick={onClickHandler}
```
Comment (shivani170) [inline src/components/userGroups/UserGroup.tsx:650]: Rename it more precisely like `onClickUserDropdownHandler`

File: src/components/userGroups/UserGroup.tsx (hunk 6 above)
Comment (Aravind-2023) [reply src/components/userGroups/UserGroup.tsx:667]: Here in this task I am supposed to make  the system user permission also as non-editable ,so I have added system in this condition.

File: src/components/userGroups/UserGroup.tsx [hunk 13]
```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect, useRef, useMemo, useCallback, useContext } from 'react'
+import React, { useState, useEffect, useRef, useMemo, useCallback, useContext, ChangeEvent } from 'react'
```
Comment (sohel-rp) [inline src/components/userGroups/UserGroup.tsx:1]: remove unused `ChangeEvent` import

File: src/components/userGroups/UserGroup.tsx [hunk 14]
```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect, useRef, useMemo, useCallback, useContext } from 'react'
+import React, { useState, useEffect, useRef, useMemo, useCallback, useContext,  } from 'react'
```
Comment (milindk8) [inline src/components/userGroups/UserGroup.tsx:1]: remove this comma

**Review by sohel-rp — APPROVED** [review]

File: src/setupProxy.js [hunk 15]
```diff
@@ -19,4 +19,4 @@ module.exports = function (app) {
             secure: false,
         }),
     )
-}
+}
```
Comment (shivani170) [inline src/setupProxy.js:22]: revert this file

**Review by milindk8 — APPROVED** [review]
